### ISSUE TITLE
Change container runtime in aws_eks_cluster module

### ISSUE THREAD

**Issue by aliartiza75** [hashicorp/terraform-provider-aws#24173]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I am using this [aws_eks_cluster](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster) module to deploy an EKS Cluster. The cluster it deploys uses **docker** as a runtime. I want to use **containerd** as container runtime.

In the documentation there is no parameter to change the container runtime.

### New or Affected Resource(s)

* EKS 1.21

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

https://kubernetes.io/blog/2020/12/02/dont-panic-kubernetes-and-docker/

Is there a way to change the container runtime in this existing module?

Thank you

**Comment by justinretzolk** [comment]
Hey @aliartiza75 👋 Thank you for taking the time to raise this! Based on [this AWS document](https://aws.amazon.com/blogs/containers/amazon-eks-1-21-released/), my read is that you can specify the container runtime as `containerd` by modifying the `userdata` of [`aws_eks_node_group` resources](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group). In looking into this, I found a couple of additional links that may be of further assistance.

- [AWS EKS: Launch Template Support](https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#launch-template-user-data)
- awslabs/amazon-eks-ami#844

**Comment by justinretzolk** [comment]
Hey @olegburca 👋 I'm always willing to be wrong (especially in this case, where EKS is certainly not my specialty), which is one of the reasons I typically try to provide answers to things like this, but ultimately leave them open in case someone else has more information and can chime in 🙂. That said, it looks like the issue you linked was before user data customization was introduced to EKS node groups ([AWS blog post here](https://aws.amazon.com/blogs/containers/introducing-launch-template-and-custom-ami-support-in-amazon-eks-managed-node-groups/), dated 17-08-2020, a couple of months after the linked issue was closed).

Regarding the managed node groups being a different thing -- I follow that those links are separate from `aws_eks_cluster`, but my read was that `aws_eks_node_groups` were indeed managed node groups, given that the documentation of that resource links to [AWS: Managed node groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html). Assuming that my read is correct, [this document](https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#launch-template-user-data) (under Specifying an AMI -> Provide user data to pass arguments to the `bootstrap.sh` file included with an Amazon EKS optimized AMI -> User data in a launch template) seems to detail the original ask: 

> Specify the following information in the user data section of your launch template. This example creates a node group that `containerd` as the runtime.

The `pre_bootstrap_user_data` argument that you mentioned looks a bit like what I saw in the [terraform-aws-modules/eks](https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest/submodules/eks-managed-node-group?tab=inputs) module -- is that perhaps where you pulled that from? If so, I wouldn't expect that argument to be accepted by the `aws_eks_node_group` resource itself.
